### PR TITLE
Document default realm and login authorization

### PR DESCRIPTION
**Pull request by greghudson**
[In a recent admin/ops call, the topic of wanting multiple default realms came up, and I thought it might be helpful to document what the default realm does and how to override it in each case.  We are also missing prose documentation for local authorization, which is one of the chief behaviors affected by the default realm.  This PR contains documentation for both.]

Add documentation to host_config.rst describing what the default realm
does.  Also add documentation discussing login authorization
configuration, and give an example showing how to give login access to
principals from a realm other than the default realm.

### CODE REVIEW

File: doc/admin/host_config.rst [hunk 1]
```diff
@@ -7,6 +7,119 @@ application servers, or KDCs, can be configured using
...
+    [realms]
+        ATHENA.MIT.EDU = {
+            auth_to_local_names = {
+                # Careful, these match principals in any realm!
+                host/example.com = hostaccount
+                fred = localfred
+            }
+        }
+
+Local authorization behavior can be modified using plugin modules; see
```
Comment (kaduk) [inline doc/admin/host_config.rst]: Maybe "can also"?

Comment (greghudson) [inline doc/admin/host_config.rst]: Sure.

File: doc/admin/host_config.rst [hunk 2]
```diff
@@ -7,6 +7,119 @@ application servers, or KDCs, can be configured using
...
+files located in account home directories, the **k5login_directory**
+relation in the :ref:`libdefaults` section can specify a directory
+containing one file per account uname.
+
+By default, if a ``.k5login`` file is present, it controls
+authorization both positively and negatively--any principal name
+contained in the file is granted access and any other principal name
+is denied access, even if it would have had access if the ``.k5login``
+file didn't exist.  The **k5login_authoritative** relation in the
+:ref:`libdefaults` section can be used to make ``.k5login`` files
```
Comment (mrogers950) [inline doc/admin/host_config.rst]: Changing to "section can be set to false to make.." may make it clearer about the k5login_authoritative default value in this paragraph.

Comment (greghudson) [inline doc/admin/host_config.rst]: Sure, I will make that change.

**Pull request merged**